### DISCUSSION TITLE
Questions about "synapse_id" in Report/Neurodamus

### DISCUSSION THREAD

**Discussion by WeinaJi** [BlueBrain/neurodamus#223] [General]
https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-1142

> At present, when running sonata simulations where two or more edge populations converge on one postsynaptic population (for instance a cortex with projections) it seems to be impossible to specify which edge populations to record synapse variables from. It also seems to be impossible to distinguish synapses from different edge populations from one another once the recording is done. At least one of these features would be very helpful at present, and a requirement for using synapse reports in future multipopulation simulations.

https://bbpteam.epfl.ch/project/issues/browse/REP-73

> Right now the synapse ids reported are read from the sonata edge file and re-indexed for every cell (0 based), while it's a continuum for the whole circuit in sonata.
The idea is to report the same global synapse ids that are in the sonata edges file.
More information about this can be found in the ticket created by András Ecker [BBPBGLIB-775](https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-775)
Some highlights:
Sirio Bolaños Puchet has implemented a mapping between the global/sonata synapse IDs and the local multi index synapse IDs. His code is in the simulation directory, and here is a version that implements the mapping for every gid in the report: https://bbpgitlab.epfl.ch/conn/simulation/sscx-analysis/-/blob/master/plasticity/map_syn_idx.py

**Comment by jorblancoa** [comment]
https://bbpteam.epfl.ch/project/issues/browse/REP-42
[[REP-42] how to read synaptic report when projection is included.pdf](https://github.com/user-attachments/files/18025493/REP-42.how.to.read.synaptic.report.when.projection.is.included.pdf)
